src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,56 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_VALID_LENGTH = 6;
+    private static final int MIN_VALID_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkLogin(user);
+        checkPassword(user);
+        checkAge(user);
+        return saveUser(user);
+    }
+
+    private void checkLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null.");
+        }
+        if (user.getLogin().isBlank() || user.getLogin().length() < 6) {
+            throw new RegistrationException(
+                    "User login should have at least 6 non-whitespace characters");
```

Comment:
Also better to use a constant for the message, if you change the constant value  you need to change the message too (check it in other messages)
```suggestion
                    "User login should have at least " + MIN_VALID_LENGTH + " non-whitespace characters");
```